04_hex_and_bytes.md

```diff
@@ -44,22 +44,33 @@ Wouldn't we expect this to look more like `00000001` so that it is correctly int
 Well, this gets us into the topic of endianness.
 
 ### Little Endian vs Big Endian
-Bytes can be stored in two different orders.
-When bytes are stored as little-endian, the least significant byte is stored first (in the lowest memory address).
-Modern CPUs usually store numbers as little endian.
-When bytes are stored as big-endian, the most significant byte is stored first.
-So when we look at a transaction, the version is defined by the protocol as being 4 bytes in *little endian*.
-This means the least significant position appears first.
-What do we mean by *position*? Well let's look at decimal numbers for comparison.
-
-When looking at decimal numbers, we typically read it from left to right with the most sigificant digit first to the least significant digit last.
+Bytes can be stored in two different orders: little vs. big endian.
+When bytes are stored as little-endian, the least significant byte is stored in the lowest memory address.
+That's how modern CPUs usually store numbers which make sense from a logical engineering perspective.
+When bytes are stored as big-endian, it's the opposite: the most significant byte is stored in the lowest memory address.
+
+In its heart, Bitcoin is a communication protocol.
+For communication protocols, the different endianness are crucial to understand because bytes will be sent in a specific sequence.
+If the receiver expects to see data in a different order from the sender, it will misinterpret data completely.
+Thus, communication protocols specify which order bytes will be sent.
+In the Bitcoin protocol, bytes are transmitted in little-endian order.
+That means that the least significant byte is sent first.
+
+So, the version 1 field will be serialized as hexadecimal `00000001` using the base 16 math as we did before.
```

Comment:
the serialization is `10000000` which is interpreted as little endian, unless i'm misunderstanding

Reply:
This is probably my pedantic professorial background kicking in, of course we can revert this to not confuse people, but let me explain.

1 = `00000001` in unsigned 32 bit, that's the model of the positional number system that became the de facto convention of having the MSB on the left as we already did in decimal the system.

Then come computers (the actual machines) and the problem of embedding this characters in a physical medium. Little-endian means "lower bytes in lower addresses", just as you defined in the text. So, if we take an array of bytes, LSB will be at index 0 and MSB at index 3.

This is still the same positional system in math. As you point out, `00 00 00 01` is to be interpreted as 0*16^3 + 0*16^2 + 0*16^1 + 1*16^0. If we understand the exponents as addresses (or indices of an array, as they truly are), conventional math is little-endian already.

The confusion comes because the convention we use to dump memory (and arrays) is the opposite of that we use for numbers.

In the positional system (MSB at left): 
```
[00, 00, 00, 01]
  ^3  ^2  ^1  ^0
```

Conventional memory dump (LSB at left): 
```
[01, 00, 00, 00]
  ^0  ^1  ^2  ^3
```

You see, there's no dichotomy between little-endian nor big-endian up to this point other than the way we print stuff in terminals. This is already the array we transmit. And it is quite natural to transmit the LSB first since this is the same way the CPU (and our minds) already traverse arrays, from index 0 (the beginning).

I'm not trying to patronize you, sorry for the long comment. While I was writing it I realized that the word `serialize` would cause confusion because I tried to use it in a "pure" form here. But it already has a conventional meaning in the bitcoin parlance (which is to conflate the memory representation with the strings we print in terminals to inspect data). Probably it is better to say we serialize something when we are referring to the strings.

Reply:
What about change this line to:

So, the version 1 filed will be represented as hexadecimal `00000001` using the base 16 math as we did before.